### PR TITLE
Fix #17351: Let external scanner parse #! cmd identifier as well ##newshell

### LINKED ISSUE #17351: newshell: # comment wrongly parsed
### Work environment

| Questions                                            | Answers
|------------------------------------------------------|--------------------
| OS/arch/bits (mandatory)                             | Irrelevant
| File format of the file you reverse (mandatory)      | Irrelevant
| Architecture/bits of the file (mandatory)            | x86_64
| r2 -v full output, **not truncated** (mandatory)     | radare2 4.6.0-git 27772 @ linux-x86-64 git.4.4.0-472-gaaa51ec0e commit: aaa51ec0ee12f9188e71800dd3c4dbd537076d2e build: 2020-07-24__10:14:53

### Expected behavior
```

```
Nothing should be printed, because `#!....` should be treated as a comment if not used as a command.

### Actual behavior
```
!ciao
```

### Steps to reproduce the behavior 
```
$ radare2 -e cfg.newshell=true -
> ?e #!ciao
```

### PR DESCRIPTION
**Pull request by ret2libc**
**Your checklist for this pull request**
- [x] I've read the [guidelines for contributing](https://github.com/radareorg/radare2/blob/master/DEVELOPERS.md) to this repository
- [x] I made sure to follow the project's [coding style](https://github.com/radareorg/radare2/blob/master/DEVELOPERS.md#code-style)
- [x] I've added tests that prove my fix is effective or that my feature works (if possible)
- [ ] I've updated the documentation and the [radare2 book](https://github.com/radareorg/radare2book) with the relevant information (if needed)

**Detailed description**

For unknown reasons, `#!` was parsed separately from other command identifiers. This PR makes `#!` identifier parsed as any other command identifier by the external scanner.

**Test plan**

tree-sitter specific tests should be green.

**Closing issues**

Closes #17351

### CODE REVIEW

**Comment by ret2libc** [comment]
I'm not sure why codecov is failing recently... I know that we disabled it by mistake some time ago and recently I re-enabled it, but it seems it's always red :/

**Comment by trufae** [comment]
+20,339 −20,954

File: shlr/radare2-shell-parser/src/scanner.c [hunk 1]
```diff
@@ -53,10 +53,16 @@ static bool is_special_start(const int32_t ch) {
...
 
-static bool is_mid_command(const char *res, const int32_t ch) {
+static bool is_mid_command(const char *res, int len, const int32_t ch) {
+	if (res[0] == '#') {
+		if (len == 1) {
+			return ch == '!' || ch == '?';
+		}
+		return ch == '?';
+	}
 	return isalnum(ch) ||  ch == '$' || ch == '?' || ch == '.' || ch == '!' ||
```
Comment (trufae) [inline shlr/radare2-shell-parser/src/scanner.c:66]: missing space

**Review by trufae — APPROVED** [review]
Didnt tested but lgtm, just a little missing space.

**Comment by trufae** [comment]
Add ## tag in the pr title

**Comment by ret2libc** [comment]
> +20,339 −20,954

Eheh yeah that's the bad of generate C code, but I've never looked at that code :) And fortunately we were able to tell github to not show the diff of those files, so it won't disturb us.

**Pull request merged**